src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/SolutionExplorer/RestoreCommandHandler.cs

```diff
@@ -47,7 +47,7 @@ private async Task RunSolutionRestoreAsync()
                 SB.IBrokeredServiceContainer serviceContainer = await _asyncServiceProvider.GetServiceAsync<SB.SVsBrokeredServiceContainer, SB.IBrokeredServiceContainer>();
                 IServiceBroker serviceBroker = serviceContainer.GetFullAccessServiceBroker();
 
-                INuGetSolutionService nugetSolutionService = await serviceBroker.GetProxyAsync<INuGetSolutionService>(NuGetServices.SolutionService);
+                INuGetSolutionService nugetSolutionService = await serviceBroker.GetProxyAsync<INuGetSolutionService>(NuGetServices.SolutionService, cancellationToken: cancellationTokenSource.Token);
```

Comment:
Maybe better to make separate PR for this change, not related to `.Any()` change.